File: Backend/sistema-gestion-elsur/src/main/java/com/elsur/sistema_gestion/models/Persona.java
package com.elsur.sistema_gestion.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "persona")
@Data
public class Persona {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_persona")
    private Integer idPersona;

    private String nombre;
    private String apellido;

    @Column(name = "numero_documento", unique = true, nullable = false)
    private String numeroDocumento;

    private String telefono;
    private String email;

    // --- RELACIONES ATÓMICAS ---

    @ManyToOne
    @JoinColumn(name = "id_tipo_documento")
    private TipoDocumento tipoDocumento;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_direccion")
    private Direccion direccion;

    @ManyToOne
    @JoinColumn(name = "id_tipo_persona")
    private TipoPersona tipoPersona;


}

File: frontend/src/types/Persona.ts
// src/types/Persona.ts
// src/types/Persona.ts
import type { Direccion } from './Direccion';
import type { TipoDocumento } from './TipoDocumento';
import type { TipoPersona } from './TipoPersona';

export interface Persona {
    idPersona?: number;
    nombre: string;
    apellido: string;
    numeroDocumento: string;
    telefono: string;
    email: string;
    tipoDocumento: TipoDocumento;
    direccion: Direccion;
    tipoPersona: TipoPersona;
}